1. Задача 1 Даны значения величины заработной платы заемщиков банка (zp) и значения их
поведенческого кредитного скоринга (ks):
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110],
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832].
Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с
помощью функции cov из numpy
Полученные значения должны быть равны.
Найдите коэффициент корреляции Пирсона с помощью ковариации и
среднеквадратичных отклонений двух признаков,
а затем с использованием функций из библиотек numpy и pandas.

In [1]:
import numpy as np

In [2]:
zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]
ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]

In [3]:
zp_ar = np.array([zp])
ks_ar = np.array([ks])
zp_ar_ks_ar = zp_ar * ks_ar
zp_ar_ks_ar

array([[ 14035,  25830, 166060, 183800,  18360,  51730,  35262, 135300,
         89520,  91520]])

In [4]:
m_xy = np.sum(zp_ar_ks_ar) / zp_ar_ks_ar.shape[1]
m_xy

81141.7

In [5]:
m_x = np.sum(zp_ar) / zp_ar.shape[1]
m_x

101.4

In [6]:
m_y = np.sum(ks_ar) / ks_ar.shape[1]
m_y

709.9

In [11]:
cov_xy = m_xy - m_x * m_y
round(cov_xy, 2)

9157.84

In [8]:
np.cov(zp_ar, ks_ar, ddof=)

array([[ 3494.64,  9157.84],
       [ 9157.84, 30468.89]])

cov = 9157.84 \
cov_xy = 9157.84

In [12]:
pirson = cov_xy / (np.std(zp_ar, ddof=0) * np.std(ks_ar, ddof=0))
pirson

0.8874900920739158

In [14]:
np.corrcoef(zp_ar, ks_ar)

array([[1.        , 0.88749009],
       [0.88749009, 1.        ]])

In [15]:
import pandas as pd

In [19]:
data = {'zp': zp, 'ks': ks}

In [20]:
df = pd.DataFrame(data)

In [25]:
df

zp   ks
0   35  401
1   45  574
2  190  874
3  200  919
4   40  459
5   70  739
6   54  653
7  150  902
8  120  746
9  110  832

In [26]:
corr_matrix = df.corr()

In [27]:
corr_matrix

zp       ks
zp  1.00000  0.88749
ks  0.88749  1.00000

2. Задача 2 Измерены значения IQ выборки студентов,
обучающихся в местных технических вузах:
131, 125, 115, 122, 131, 115, 107, 99, 125, 111.
Известно, что в генеральной совокупности IQ распределен нормально.
Найдите доверительный интервал для математического ожидания с надежностью 0.95.

In [29]:
x = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])
x_mean = np.mean(x)
x_mean

118.1

In [30]:
from scipy import stats

In [31]:
t = stats.t.ppf((1-0.95)/2, len(x) - 1)
t

-2.2621571627409915

In [32]:
x_mean - t * (np.std(x, ddof=1)/np.sqrt(len(x)))

125.64391634841274

In [33]:
x_mean + t * (np.std(x, ddof=1)/np.sqrt(len(x)))

110.55608365158724

Ответ: (110.55608365158724, 125.64391634841274)

3. Задача 3 Известно, что рост футболистов в сборной распределен нормально
с дисперсией генеральной совокупности, равной 25 кв.см. Объем выборки равен 27,
среднее выборочное составляет 174.2. Найдите доверительный интервал для
математического
ожидания с надежностью 0.95.

In [35]:
z =  stats.norm.ppf(0.05)
z

-1.6448536269514729

In [36]:
174.2 - z * np.sqrt(25/27)

175.7827611404966

In [37]:
174.2 + z * np.sqrt(25/27)

172.61723885950337

Ответ: (172.61723885950337, 175.7827611404966)